src/lib.rs

```diff
@@ -385,6 +394,73 @@ impl Interpolate for Quaternion<f32> {
   }
 }
 
+#[cfg(feature = "impl-nalgebra")]
+impl<N : Scalar, D : DimName> Interpolate for na::Point<N, D> 
+  where DefaultAllocator : Allocator<N, D>,
+        <DefaultAllocator as Allocator<N, D>>::Buffer: Copy,
+        N : Interpolate,
+{
+  fn lerp(a: Self, b: Self, t: f32) -> Self {
+    let lerp = |c1 : N, c2 : N| { Interpolate::lerp(c1, c2, t) };
+    let coords = na::Matrix::zip_map(&a.coords, &b.coords, lerp);
```

Comment:
Can you please explain what’s happening here? You’re implementing `Interpolate::lerp` on N-dimensional points and you use matrices?

Reply:
This comes from the fact that the Point type is a struct with a single field, coord, which is a VectorN, and the VectorN type is ultimately a specialization of the Matrix type.

Looking at it, I realized I could use Vector::zip_map, which is clearer because the type of 'coord' is Vector. I can also do that for the Vector impls.